helpers/lib/common.js

```diff
@@ -0,0 +1,38 @@
+function isObject(val) {
+    return val !== null && typeof val === 'object';
+}
+
+function isOptions(val) {
+    return isObject(val) && val.fn;
+}
+
+function isString(val) {
+    return typeof val === 'string';
+}
+
+function isEmptyObject(val) {
+    return isObject(val) && Object.keys(val).length === 0;
+}
+
+function isTruthy(val) {
+    if (Array.isArray(val)) {
+        return !!val.length;
+    } else if (isEmptyObject(val)) {
+        return false;
+    } else {
+        return !!val;
+    }
+}
+
+function sameValueZero(val1, val2) {
```

Comment:
I don't understand what this is trying to do. We probably need a comment here and probably a function name change.

Reply:
sameValueZero is a Lodash algorithm that also allows for NaN === NaN to return true.

Reply:
changed.